lib/_http_outgoing.js

```diff
@@ -296,21 +296,20 @@ function _storeHeader(firstLine, headers) {
     header: firstLine
   };
 
-  var key;
-  if (headers === this[outHeadersKey]) {
-    for (key in headers) {
-      const entry = headers[key];
-      processHeader(this, state, entry[0], entry[1], false);
-    }
-  } else if (Array.isArray(headers)) {
-    for (var i = 0; i < headers.length; i++) {
-      const entry = headers[i];
-      processHeader(this, state, entry[0], entry[1], true);
-    }
-  } else if (headers) {
-    for (key in headers) {
-      if (hasOwnProperty(headers, key)) {
-        processHeader(this, state, key, headers[key], true);
+  if (headers) {
+    if (headers === this[outHeadersKey]) {
+      for (const entry of Object.values(headers)) {
```

Comment:
The former was likely better performance wise. AFAIK `Object.values()` is still relatively slow.

Reply:
From what I tested, looks `Object.values` is a litter faster in v8 `7.0`.
```
ObjectValuesUseFor: 0.303ms
ObjectValues: 0.272ms
```

Test case: 
```js
const TEST_COUNT = Math.pow(10, 2);

const pojo = {
    name: 'n',
    age: 10
}

function Benckmark(desc, func) {
    console.time(desc)
    let sum = 0
    for (let index = 0; index < TEST_COUNT; index++) {
        func()
    }
    console.timeEnd(desc)
}

const noop = () => {}

function ObjectValues(o) {
    for (const key of Object.values(o)) {	
        noop(key)
    }
}

function ObjectValuesUserFor(o) {
    for (const key in o) {	
        const v = o[key]
        noop(v)
    }
}
```

Reply:
Be careful with this kind of benchmark, V8 could be optimizing away the loops entirely because of the noop function.

Reply:
I test on my win10 after change `noop` to `const noop = (i) => {console.log(i)}`, the perf is basically the same.

Env:
```
  System:
    OS: Windows 10
    CPU: (8) x64 Intel(R) Core(TM) i7-4720HQ CPU @ 2.60GHz
    Memory: 4.56 GB / 15.89 GB
  Binaries:
    Node: 11.10.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.13.0 - C:\Program Files\nodejs\yarn.CMD
    npm: 5.6.0 - C:\Program Files\nodejs\npm.CMD
```